src/odemis/gui/cont/tabs.py

```diff
@@ -2000,6 +2008,116 @@ def __init__(self, name, button, panel, main_frame, main_data):
         self.tb.add_tool(TOOL_ACT_ZOOM_FIT, self.view_controller.fitViewToContent)
         # TODO: add tool to zoom out (ie, all the scintillators, calling canvas.zoom_out())
 
+    def _on_btn_optical_autofocus(self, _):
+        """
+        Start or cancel the Optical Autofocus calibration when the button is triggered.
+        """
+        # check if cancelled
+        if self.tab_data.is_calibrating.value:
+            fastem._executor.cancel()
+            return
+
+        # calibrate
+        self.tab_data.is_calibrating.unsubscribe(self._on_is_acquiring)
+        # Don't catch this event (is_calibrating = True) - this would disable the button,
+        # but it should be still enabled in order to be able to cancel the calibration
+        self.tab_data.is_calibrating.value = True  # make sure the acquire/tab buttons are disabled
+        self.tab_data.is_calibrating.subscribe(self._on_is_acquiring)
+        logging.debug("Starting Optical Autofocus calibration")
+        # Start alignment
+        f = fastem.align(self.tab_data.main.ebeam, self.tab_data.main.multibeam,
+                               self.tab_data.main.descanner, self.tab_data.main.mppc,
+                               self.tab_data.main.stage, self.tab_data.main.ccd,
+                               self.tab_data.main.beamshift, self.tab_data.main.det_rotator,
+                               calibrations=[Calibrations.OPTICAL_AUTOFOCUS])
+        f.add_done_callback(self._on_calibration_done)  # also handles cancelling and exceptions
+        # connect the future to the progress bar and its label
+        self._future_connector = ProgressiveFutureConnector(f, wx.Gauge(), full=False)
```

Comment:
Where will this wx.Gauge() be shown? Typically it's created at init (via the main.xrc), and shown/hidden. Is that something you will implement in a separate PR? If so, please mark as a comment.

Also, the `full` only matters if you pass a `label`, so here it should be fine to drop.

Reply:
The `Optical Autofocus` calibration in the Overview tab does not need a guage to be shown as per the mock-up. I can create a gauge and then hide it if you say so @pieleric.

Reply:
The only point of a `ProgressiveFutureConnector` is to connect the gauge (and text) to the (Progressive) future. If you don't show a gauge, you don't need the connector at all.